modules/core/src/main/java/org/apache/synapse/aspects/flow/statistics/data/raw/BasicStatisticDataUnit.java

```diff
@@ -102,4 +109,12 @@ public boolean isOutOnlyFlow() {
 	public void setIsOutOnlyFlow(boolean isOutOnlyFlow) {
 		this.isOutOnlyFlow = isOutOnlyFlow;
 	}
+
+	public void collectMetadataForElastic(MessageContext messageContext) {
```

Comment:
It seems we are just initiating the object here and nothing seems to be collected. Shall we give a meaningful name to the method?

Reply:
How about `generateElasticMetadata` ?

Reply:
+1

Reply:
Fixed